src/robot/output/listeners.py

```diff
@@ -221,15 +221,21 @@ class ListenerProxy(AbstractLoggerProxy):
                 'output_file', 'report_file', 'log_file', 'debug_file',
                 'xunit_file', 'close']
 
-    def __init__(self, name, args):
-        listener = self._import_listener(name, args)
-        AbstractLoggerProxy.__init__(self, listener)
-        self.name = name
-        self.version = self._get_version(listener)
+    def __init__(self, listener, args):
+        listener_instance = self._get_listener(listener, args)
+        AbstractLoggerProxy.__init__(self, listener_instance)
+        self.name = str(listener)
+        self.version = self._get_version(listener_instance)
         if self.version == 1:
             LOGGER.warn("Listener '%s' uses deprecated API version 1. "
                         "Switch to API version 2 instead." % self.name)
 
+    def _get_listener(self, listener, args):
+        if isinstance(listener, str):
```

Comment:
Should check against `basetring` to allow passing listener name as `str` and `unicode`.